### ISSUE TITLE
Missing index for ``SELECT * FROM `*PREFIX*polls_polls` WHERE (`deleted` = 0) OR (`owner` = "admin")``

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/polls#3361]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### What went wrong, what did you observe?

I'm analyzing a production installation of this app and noticed that the query `SELECT * FROM `*PREFIX*polls_polls` WHERE (`deleted` = 0) OR (`owner` = "admin")` does not use an index.

### What did you expect, how polls should behave instead?

Depending on the data the use of an index might be beneficial or not. E.g. if you only have polls of one user, then a full table scan be will more efficient anyway.

An possible index could be `create index polls_polls_owner_deleted_idx on oc_polls_polls (owner, deleted);`.

### What steps does it need to replay this bug?

1. Install 5.x

### Installation method

Installed/updated from the appstore (Apps section of your site)

### Installation type

First time installation

### Affected polls version

5.4.3

### Which browser did you use, when experiencing the bug?

- [ ] Firefox
- [ ] Chrome
- [ ] Chromium/Chromium based (i.e. Edge)
- [ ] Safari
- [ ] Other/Don't know

### NC version

Nextcloud 26

### PHP engine version

PHP 8.0

### Database engine

MySQL

### Which user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other/Don't know

**Comment by dartcafe** [comment]
Yes true, but this should be an enhancement rather than a bug, right?
